include/oneapi/dpl/internal/binary_search_impl.h

```diff
@@ -128,13 +132,30 @@ lower_bound_impl(Policy&& policy, InputIterator1 start, InputIterator1 end, Inpu
     auto keep_result = oneapi::dpl::__ranges::__get_sycl_range<__bknd::access_mode::read_write, OutputIterator>();
     auto result_buf = keep_result(result, result + value_size);
     auto zip_vw = make_zip_view(input_buf.all_view(), value_buf.all_view(), result_buf.all_view());
-    __bknd::__parallel_for(::std::forward<Policy>(policy),
-                           custom_brick<StrictWeakOrdering, decltype(size), lower_bound>{comp, size}, value_size,
-                           zip_vw)
-        .wait();
+    // Enable index calculation to proceed with uint32_t if input range is small enough.
+    if (size <= ::std::numeric_limits<::std::uint32_t>::max())
+    {
+        __bknd::__parallel_for(::std::forward<Policy>(policy),
```

Comment:
The code in these two branches looks like the same excepting used policy.
May we combine it into one branch?

Reply:
I don't think it is possible here. In the first branch `custom_brick` uses a `::std::uint32_t` index type and a `::std::size_t` in the second branch.  We have to make the dispatch decision at runtime to choose the correct kernel.